unit-tests/src/test/java/org/eclipse/collections/impl/multimap/AbstractMutableMultimapTestCase.java

```diff
@@ -197,6 +197,28 @@ public void putAllFromIterable()
         Assert.assertEquals(this.newMultimapWithKeysValues(1, "One", 2, "Two", 1, "Three", 1, "Four"), multimap);
     }
 
+    @Test
+    public void getIfAbsentPut()
+    {
+        MutableMultimap<Integer, Integer> multimap = this.newMultimap();
+        Assert.assertFalse(multimap.containsKey(1));
+        Assert.assertEquals(0, multimap.size());
+        Assert.assertEquals(this.createCollection(1), multimap.getIfAbsentPut(1, Lists.mutable.with(1)));
+        Verify.assertThrows(UnsupportedOperationException.class, () -> multimap.getIfAbsentPut(1, Lists.mutable.with(1)).add(1));
+
+        multimap.putAll(2, Lists.mutable.with(2, 2));
+        multimap.putAll(3, Lists.mutable.with(3, 3, 3));
+        Assert.assertEquals(this.createCollection(1), multimap.getIfAbsentPut(1, Lists.mutable.empty()));
+        Assert.assertEquals(this.createCollection(2, 2), multimap.getIfAbsentPut(2, Lists.mutable.empty()));
+        Assert.assertEquals(this.createCollection(3, 3, 3), multimap.getIfAbsentPut(3, Lists.mutable.empty()));
+        Assert.assertEquals(this.createCollection(4, 4, 4, 4), multimap.getIfAbsentPut(4, Lists.mutable.with(4, 4, 4, 4)));
+        Assert.assertEquals(4, multimap.sizeDistinct());
+        int multimapSize = this.createCollection(1).size() + this.createCollection(2, 2).size() + this.createCollection(3, 3, 3).size() + this.createCollection(4, 4, 4, 4).size();
+        Assert.assertEquals(multimapSize, multimap.size());
+
+        Verify.assertThrows(UnsupportedOperationException.class, () -> multimap.getIfAbsentPut(5, Lists.mutable.with(5)).add(5));
```

Comment:
Add some assertions about adding empty values? Preferably early, so if anything is messed up the size checks have a chance of catching the problem.